js/src/types.ts

```diff
@@ -47,21 +48,21 @@ export type MessageTypes = {
 }
 
 export interface MessageMetadata {
-  id: string
-  threadId: string
+  threadId: string // technically this has to be an rfq id because that's how you start a thread
   parentId?: string
   from: string
   to: string
   createdTime: string
 }
 
 export type TbDEXMessage<T extends keyof MessageTypes> = MessageMetadata & {
+  id: TypeID<T>
```

Comment:
ok so with this change, id is still a "top level" tbdex message field, it's just that it is an ID of type T using typeid instead of our own :)

Reply:
Correct! the message itself still looks the same and the interface for constructing a message is the same